R/pmcmc_tools.R

```diff
@@ -194,3 +194,108 @@ combine_state <- function(x) {
   ret$state <- state
   ret
 }
+
+
+##' Convert sampled trajectories from [pmcmc()] to a Tidy (long) format,
+##' and optionally marginalise over particles to produce summary statistics.
+##' Helper function useful for ggplotters.
+##'
+##' @title Convert sampled trajectories from [pmcmc()] to a Tidy (long) format
+##' @param object Results of running [pmcmc()].
+##'
+##' @param summarise Logical indicating whether the trajectories should be
+##' marginalised over particles to produce summary statistics. Default is
+##' `FALSE`.
+##'
+##' @param quantiles Optional vector giving quantiles to output if `summarise`
+##' is TRUE defaults to c(0.025, 0.5, 0.975) corresponding to 2.5-, 50-, and
+##' 97.5-percentiles
+##'
+##' @param summary_function Optional function to calculated user-defined
+##' summary statistics - defaults to calculating mean, 2.5-, 50-, and
+##' 97.5-percentiles
+##'
+##' @export
+##' @importFrom stats quantile
+
+pmcmc_tidy_trajectories <- function(object, summarise = FALSE,
+                                    quantiles = c(0.025, 0.5, 0.975),
+                                    summary_function = NULL) {
```

Comment:
Thinking about the interface here, perhaps:

````
pmcmc_tidy_trajectories <- function(object, summarise = FALSE, summary = NULL) {
  if (is.null(summary)) {
    summary <- function(x) {
      quantiles <- c(0.025, 0.5, 0.975)
      ret <- c(mean(x), quantile(x, quantiles, names = FALSE))
      set_names(ret, c("mean", paste0("q", quantiles * 100)))
    }
  }
```

then we document that this is the default. Otherwise we're holding a weird interface.

If the user does provide a function, it needs to take a single argument and return a named numeric vector